app/services/katello/applicability/applicable_content_helper.rb

```diff
@@ -45,6 +45,22 @@ def fetch_errata_content_ids
         return Katello::Erratum.find_by_sql([query, {content_facet_id: content_facet.id, repo_ids: self.bound_library_instance_repos} ]).map(&:id).uniq
       end
 
+      def fetch_module_stream_content_ids
+        # Query for all applicable module stream ids
+        query = 'select katello_repository_module_streams.module_stream_id as id from katello_repository_module_streams
+                     inner join katello_module_stream_rpms
+                        on  katello_repository_module_streams.module_stream_id  = katello_module_stream_rpms.module_stream_id
+                     inner join katello_rpms
+                        on katello_rpms.id = katello_module_stream_rpms.rpm_id
+                     inner join katello_content_facet_applicable_rpms on
+                         katello_content_facet_applicable_rpms.rpm_id = katello_rpms.id
+                     where
+                          "katello_content_facet_applicable_rpms"."content_facet_id" = :content_facet_id
+                     AND katello_repository_module_streams.repository_id IN (:repo_ids)'
+
+        return Katello::ModuleStream.find_by_sql([query, {content_facet_id: content_facet.id, repo_ids: self.bound_library_instance_repos} ]).map(&:id).uniq
```

Comment:
I think SELECT DISTINCT may be faster than making Ruby do it with `.uniq`?  Also Active Record has a `.distinct` method

Reply:
I can change this and the errata method to distinct, sure